examples/basics/visuals/line_draw.py

```diff
@@ -0,0 +1,150 @@
+    def print_mouse_event(self, event, what):
+        print('%s - pos: %r, button: %s,  delta: %r' %
+              (what, event.pos, event.button, event.delta))
+
+    # get line point close to mouse pointer and its index
+    # @param: radius   - max. distance in pixels between mouse and line point to be accepted
+    # @return point, index  (numpy.array, int)
+    def select_point(self, event, radius = 5):
+        # position in scene/document coordinates
+        pos_scene = event.pos[:3]
+
+        # project mouse radius from screen coordinates to document coordinates
+        mouse_radius = (event.visual_to_canvas.imap(np.array([radius,radius,radius]))-event.visual_to_canvas.imap(np.array([0,0,0])))[0]
+        #print("Mouse radius in document units: ", mouse_radius)
+
+        # find first point within mouse_radius
+        index=0
+        for p in self.pos:
+            if np.linalg.norm(pos_scene - p) < mouse_radius:
+                #print p, index
+                # point found, return point and its index
+                return p, index
+            index +=1
+        # no point found, return None
+        return None, -1
+
+    # update marker colors, and highlight a selected marker with a given color
+    def update_markers(self, selected_index = -1, highlight_color=(1,0,0,1)):
+        self.marker_colors.fill( 1)
+        # default shape (non-highlighted)
+        shape="o"
+        size=6
+        if selected_index >=0 and selected_index <len(self.marker_colors):
+            self.marker_colors[selected_index] = highlight_color
+            # if there is a highlighted marker, change all marker shapes to a square
+            shape="s"
+            size=8
+        self.markers.set_data(pos=self.pos, symbol=shape,edge_color='red', size=size, face_color=self.marker_colors)
+
+
+    def on_mouse_press(self, event):
+        self.print_mouse_event(event, 'Mouse press')
+        pos_scene = event.pos[:3]
+
+        # find closest point to mouse and select it
+        self.selected_point, self.selected_index = self.select_point(event)
+
+        # if no point was clicked add a new one
+        if self.selected_point == None:
+            print("adding point", len(self.pos))
+            self._pos = np.append(self.pos, [pos_scene], axis=0)
+            self.set_data(pos=self.pos)
+            self.marker_colors = np.ones((len(self.pos), 4), dtype=np.float32)
+            self.selected_point = self.pos[-1]
+            self.selected_index=len(self.pos)-1
+
+        # update markers and highlights
+        self.update_markers(self.selected_index)
+
+    def on_mouse_release(self, event):
+        self.print_mouse_event(event, 'Mouse release')
+        self.selected_point = None
+        self.update_markers()
+
+    def on_mouse_move(self, event):
+        # left mouse button
+        if event.button == 1:
+            # self.print_mouse_event(event, 'Mouse drag')
+            if self.selected_point != None:
```

Comment:
you should generally use `is` for comparisions to `None`, like `if self.selected_point is not None:` here